detect_secrets/util/inject.py

```diff
@@ -13,11 +13,12 @@ def inject_variables_into_function(func: SelfAwareCallable, **kwargs: Any) -> Op
         for key in (variables_to_inject & func.injectable_variables)
     }
 
-    if set(values.keys()) != func.injectable_variables:
+    try:
+        return func(**values)
+    except TypeError:
+        # When parameters are required (and not supplied), it raises a TypeError.
         return None
```

Comment:
guess it's a little late for this but: why swallow the TypeError? if you don't call a function correctly it seems reasonable for that to raise TypeError, returning `None` seems a bit more surprising especially if the underlying function type signature doesn't even have `None` as an allowed return value

Reply:
and this is way off topic but it might be possible to use mypy generics to get rid of that `-> Optional[Any]`?

Reply:
Never too late -- the original PR is much too big, and there's no way that it could have received detailed feedback. Thus, I still appreciate the comments, even though it is "after the fact".

I originally wrote the dependency injection framework to power the filter infrastructure, so there are assumptions built in reflective of this original use case. However, it seems that we are now using it for more things than just filters, so I've made the appropriate changes and handled these exceptions at the caller level.